From 52571d65dfb945246b53af5b857aca062a69a42c Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:23:47 -0400
Subject: [pci] aer: Add a TLP header print helper

Message-id: <20140719042346.19438.60557.stgit@gir.stowe>
Patchwork-id: 85972
O-Subject: [RHEL7.1 PATCH 10/81] PCI/AER: Add a TLP header print helper
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: fab4c256a58bcd6877bca5b6afc97afa3b6664ac

commit fab4c256a58bcd6877bca5b6afc97afa3b6664ac
Author: Borislav Petkov <bp@suse.de>
Date:   Fri Dec 13 14:39:56 2013 -0700

    PCI/AER: Add a TLP header print helper

    ... and call it instead of duplicating the large printk format
    statement.

    No functionality change.

    Signed-off-by: Borislav Petkov <bp@suse.de>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/pcie/aer/aerdrv_errprint.c b/drivers/pci/pcie/aer/aerdrv_errprint.c
index 2c7c9f5..32efc5e 100644
--- a/drivers/pci/pcie/aer/aerdrv_errprint.c
+++ b/drivers/pci/pcie/aer/aerdrv_errprint.c
@@ -124,6 +124,21 @@ static const char *aer_agent_string[] = {
  "Transmitter ID"
 };
 
+static void __print_tlp_header(struct pci_dev *dev,
+          struct aer_header_log_regs *t)
+{
+ unsigned char *tlp = (unsigned char *)&t;
+
+ dev_err(&dev->dev, "  TLP Header:"
+  " %02x%02x%02x%02x %02x%02x%02x%02x"
+  " %02x%02x%02x%02x %02x%02x%02x%02x\n",
+  *(tlp + 3), *(tlp + 2), *(tlp + 1), *tlp,
+  *(tlp + 7), *(tlp + 6), *(tlp + 5), *(tlp + 4),
+  *(tlp + 11), *(tlp + 10), *(tlp + 9),
+  *(tlp + 8), *(tlp + 15), *(tlp + 14),
+  *(tlp + 13), *(tlp + 12));
+}
+
 static void __aer_print_error(struct pci_dev *dev,
          struct aer_err_info *info)
 {
@@ -178,17 +193,8 @@ void aer_print_error(struct pci_dev *dev, struct aer_err_info *info)
 
   __aer_print_error(dev, info);
 
-  if (info->tlp_header_valid) {
-   unsigned char *tlp = (unsigned char *) &info->tlp;
-   dev_err(&dev->dev, "  TLP Header:"
-    " %02x%02x%02x%02x %02x%02x%02x%02x"
-    " %02x%02x%02x%02x %02x%02x%02x%02x\n",
-    *(tlp + 3), *(tlp + 2), *(tlp + 1), *tlp,
-    *(tlp + 7), *(tlp + 6), *(tlp + 5), *(tlp + 4),
-    *(tlp + 11), *(tlp + 10), *(tlp + 9),
-    *(tlp + 8), *(tlp + 15), *(tlp + 14),
-    *(tlp + 13), *(tlp + 12));
-  }
+  if (info->tlp_header_valid)
+   __print_tlp_header(dev, &info->tlp);
  }
 
  if (info->id && info->error_dev_num > 1 && info->id == id)
@@ -250,18 +256,10 @@ void cper_print_aer(struct pci_dev *dev, int cper_severity,
  if (aer_severity != AER_CORRECTABLE)
   dev_err(&dev->dev, "aer_uncor_severity: 0x%08x\n",
          aer->uncor_severity);
- if (tlp_header_valid) {
-  const unsigned char *tlp;
-  tlp = (const unsigned char *)&aer->header_log;
-  dev_err(&dev->dev, "aer_tlp_header:"
-   " %02x%02x%02x%02x %02x%02x%02x%02x"
-   " %02x%02x%02x%02x %02x%02x%02x%02x\n",
-   *(tlp + 3), *(tlp + 2), *(tlp + 1), *tlp,
-   *(tlp + 7), *(tlp + 6), *(tlp + 5), *(tlp + 4),
-   *(tlp + 11), *(tlp + 10), *(tlp + 9),
-   *(tlp + 8), *(tlp + 15), *(tlp + 14),
-   *(tlp + 13), *(tlp + 12));
- }
+
+ if (tlp_header_valid)
+  __print_tlp_header(dev, &aer->header_log);
+
  trace_aer_event(dev_name(&dev->dev), (status & ~mask),
    aer_severity);
 }
-- 
1.7.1